### ISSUE TITLE
crdb_internal.list_sql_keys_in_range() has inconsistent output

### ISSUE THREAD

**Issue by smcvey** [cockroachdb/cockroach#87703]
There is inconsistency in the way `crdb_internal.list_sql_keys_in_range()` displays its output.  Sometimes it appears encoded in hex, other times it does not.

Reproduction, create a simple table with three string columns, and insert some data:

```
CREATE TABLE public.mytable (
    id UUID NOT NULL,
    column_a STRING NULL,
    column_b STRING NULL,
    column_c STRING NULL,
    CONSTRAINT mytable_pkey PRIMARY KEY (id ASC)
);

INSERT INTO mytable SELECT gen_random_uuid(), 'text for column a', 'text for column b', 'text for column c' FROM generate_series(1, 10);
```

Now if I use `crdb_internal.list_sql_keys_in_range()` to list the keys, I get this:

```
root@:26257/defaultdb> show ranges from table mytable;
  start_key | end_key | range_id |       range_size_mb       | lease_holder | lease_holder_locality | replicas | replica_localities
------------+---------+----------+---------------------------+--------------+-----------------------+----------+---------------------
  NULL      | NULL    |       45 | 0.00097200000000000000000 |            1 |                       | {1}      | {}
(1 row)


root@:26257/defaultdb> select * from crdb_internal.list_sql_keys_in_range(45);
                                   key                                  |                                                            value
------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0                 | 0c9b7bd50a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/0        | bcbe7b060a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/0             | 174ddc2c0a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/0                  | 9609620f0a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/0              | b32989ac0a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/0 | ff62d01a0a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/0               | 939889cb0a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/0          | 2ba38fb00a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/0                | 84e8c5740a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/0      | a2c791d90a26117465787420666f7220636f6c756d6e206116117465787420666f7220636f6c756d6e206216117465787420666f7220636f6c756d6e2063
(10 rows)
```

It's all encoded in hex, so I can use the following to decode it:

```
root@:26257/defaultdb> select key, decode(value, 'hex') from crdb_internal.list_sql_keys_in_range(45);
                                   key                                  |                                           decode
------------------------------------------------------------------------+---------------------------------------------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0                 | \f\x9b{\xd5\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/0        | \xbc\xbe{\x06\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/0             | \x17M\xdc,\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/0                  | \x96\tb\x0f\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/0              | \xb3)\x89\xac\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/0 | \xffb\xd0\x1a\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/0               | \x93\x98\x89\xcb\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/0          | +\xa3\x8f\xb0\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/0                | \x84\xe8\xc5t\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/0      | \xa2\xc7\x91\xd9\n&\x11text for column a\x16\x11text for column b\x16\x11text for column c
(10 rows)
```

So far, so good.  Now, let's move one of my columns to another column family:

```
root@:26257/defaultdb> ALTER TABLE mytable ADD COLUMN new_column string CREATE FAMILY "new_family" NOT VISIBLE AS (column_b) STORED;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable ALTER COLUMN new_column DROP STORED, ALTER COLUMN new_column SET VISIBLE, RENAME COLUMN column_b TO old_column, RENAME COLUMN new_column TO column_b,  ALTER COLUMN old_column SET NOT VISIBLE;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable DROP COLUMN old_column;
ALTER TABLE
```

Let's try to run the `crdb_internal.list_sql_keys_in_range()` function again:

```
root@:26257/defaultdb> select key, decode(value, 'hex') from crdb_internal.list_sql_keys_in_range(45);
                           key                          |                          decode
--------------------------------------------------------+-----------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0 | \xc3\xae@V\n&\x11text for column a&\x11text for column c
(1 row)
(error encountered after some results were delivered)
ERROR: decode(): encoding/hex: invalid byte: U+0022 '"'
```

Looks like it broke on one of the lines - let's remove the decoding and look at the raw output:

```
root@:26257/defaultdb> select * from crdb_internal.list_sql_keys_in_range(45);
                                    key                                   |                                         value
--------------------------------------------------------------------------+-----------------------------------------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0                   | c3ae40560a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/1/1                 | "text for column b"
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/0          | 34a173220a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/1/1        | "text for column b"
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/0               | 9e178e660a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/1/1             | "text for column b"
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/0                    | 66041e0f0a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/1/1                  | "text for column b"
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/0                | e836bd820a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/1/1              | "text for column b"
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/0   | 2f58162d0a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/1/1 | "text for column b"
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/0                 | f2dc1a300a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/1/1               | "text for column b"
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/0            | b76355460a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/1/1          | "text for column b"
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/0                  | 24797e5d0a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/1/1                | "text for column b"
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/0        | b83e43710a26117465787420666f7220636f6c756d6e206126117465787420666f7220636f6c756d6e2063
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/1/1      | "text for column b"
(20 rows)
```

Look at that - the second column family isn't in an encoded format.  But there's more inconsistency to follow.  Let's move a second column to the same column family:

```
root@:26257/defaultdb> ALTER TABLE mytable ADD COLUMN new_column string FAMILY "new_family" NOT VISIBLE AS (column_c) STORED;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable ALTER COLUMN new_column DROP STORED, ALTER COLUMN new_column SET VISIBLE, RENAME COLUMN column_c TO old_column, RENAME COLUMN new_column TO column_c,  ALTER COLUMN old_column SET NOT VISIBLE;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable DROP COLUMN old_column;
ALTER TABLE

root@:26257/defaultdb> select key, decode(value, 'hex') from crdb_internal.list_sql_keys_in_range(45);
                                    key                                   |                              decode
--------------------------------------------------------------------------+--------------------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0                   | \x1c\x9b\x9a\xb8\n&\x11text for column a
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/1/1                 | O\xcd\x96C\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/0          | \xc4\xf5\r\xb7\n&\x11text for column a
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/1/1        | \x18R,U\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/0               | A\x12(B\n&\x11text for column a
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/1/1             | iI\x1f!\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/0                    | \xeb\x8e\xc4\x10\n&\x11text for column a
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/1/1                  | ]\xdf\xd5M\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/0                | \x10\xc5<4\n&\x11text for column a
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/1/1              | \xceX\x18s\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/0   | \x99\xe1\x06y\n&\x11text for column a
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/1/1 | \x88\xf6\xc8\xa1\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/0                 | #\x87\x92\xcb\n&\x11text for column a
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/1/1               | \xeb- t\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/0            | \xe3\x0e\x8a6\n&\x11text for column a
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/1/1          | R\x0eK2\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/0                  | <\x037_\n&\x11text for column a
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/1/1                | \xd8\xf8\x98\xf5\nV\x11text for column b\x16\x11text for column c
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/0        | l\xb7\xe0\xa9\n&\x11text for column a
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/1/1      | \xea\xbc\xb3+\nV\x11text for column b\x16\x11text for column c
(20 rows)
```

Now it's all decodable again.  Let's move that second column back to the original column family:

```
root@:26257/defaultdb> ALTER TABLE mytable ADD COLUMN new_column string FAMILY "primary" NOT VISIBLE AS (column_c) STORED;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable ALTER COLUMN new_column DROP STORED, ALTER COLUMN new_column SET VISIBLE, RENAME COLUMN column_c TO old_column, RENAME COLUMN new_column TO column_c,  ALTER COLUMN old_column SET NOT VISIBLE;
ALTER TABLE

root@:26257/defaultdb> ALTER TABLE mytable DROP COLUMN old_column;
ALTER TABLE
```

However, this time, it's still all decodable:

```
root@:26257/defaultdb> select key, decode(value, 'hex') from crdb_internal.list_sql_keys_in_range(45);
                                    key                                   |                             decode
--------------------------------------------------------------------------+-----------------------------------------------------------------
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/0                   | \xbb`\xe1`\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\t\x00\xdafV\xa4F+\x9d\xfe\vW\x159|k"/1/1                 | \x0f\x181\xcc\nV\x11text for column b
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/0          | Lo\xd2\x14\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\x11>\xb3\\z\x80C\x14\xb4\a\x14,T\xd9\xe9\xd1"/1/1        | \x9f\x7fB\xca\nV\x11text for column b
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/0               | \xe6\xd9/P\n&\x11text for column aV\x11text for column c
  /Table/104/1/"B\xceZ\xe9`4I\xb5\xa1\x989\xf0\x9db9\x93"/1/1             | R-\xa3|\nV\x11text for column b
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/0                    | \x1e\xca\xbf9\n&\x11text for column aV\x11text for column c
  /Table/104/1/"V\xa4\xd8nѺCľ\x1at\xafR\x0e\x87\x84"/1/1                  | \xd7\xe2\x0fH\nV\x11text for column b
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/0                | \x90\xf8\x1c\xb4\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\x86\x9a\x12R=nCٍ\xe6\xcfg\t\xe9\x8e\x16"/1/1              | \xeb\x1a\xa0a\nV\x11text for column b
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/0   | W\x96\xb7\x1b\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\x90\x82\xb6\x9c\xff\xc2Ex\xa7\x88\x83M=\xa6\x99\xcc"/1/1 | \xe3\xfe\xb9v\nV\x11text for column b
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/0                 | \x8a\x12\xbb\x06\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\xb9k\x9d4\x10wLI\xa0\rU\xa7$j\x00\xaa"/1/1               | \xc6+\rB\nV\x11text for column b
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/0            | \xcf\xad\xf4p\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\xc31ztM\xcaN\xa6\xb9i\xa8\xdb\xe7b\x95\xc9"/1/1          | \x05\xe7\n\xfb\nV\x11text for column b
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/0                  | \\\xb7\xdfk\n&\x11text for column aV\x11text for column c
  /Table/104/1/"ʨD\xde\xfeRCH\xb1}\xe6\u0082:\xe0\x8b"/1/1                | 1V\xde\xba\nV\x11text for column b
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/0        | \xc0\xf0\xe2G\n&\x11text for column aV\x11text for column c
  /Table/104/1/"\xf2\xfb\x8e\xc4h\xb7I\xa8\x90\nb\xfdo\xe0\x8ex"/1/1      | e\xd8=D\nV\x11text for column b
(20 rows)
```

It seems to only break after moving the first column across, but further modifications are fine.

Jira issue: CRDB-19491

Epic CRDB-28089

**Comment by rafiss** [comment]
Moving to the multitenant board since this was last touched in b3cd165f0a4602845ad94d81a9625f3708694187

**Comment by shubhamdhama** [comment]
TLDR; This issue seems to be fixed by the commit linked above https://github.com/cockroachdb/cockroach/commit/b3cd165f0a4602845ad94d81a9625f3708694187. And I'm not able to reproduce it either.

Here is my analysis:

When we changed `kv.KeyValue` with `roachpb.KeyValue` here https://github.com/cockroachdb/cockroach/commit/b3cd165f0a4602845ad94d81a9625f3708694187#diff-ecbfff3cc85f94442fdb6b3a6a68629aaba8556cf0b96c7fb47dbe97e9ac0323L1742-L1747, we had to replace `kv.PrettyValue` with `kv.Value.PrettyPrint`. As you might have figured, these functions are used for `list_sql_keys_in_range`.
 https://github.com/cockroachdb/cockroach/commit/b3cd165f0a4602845ad94d81a9625f3708694187#diff-ecbfff3cc85f94442fdb6b3a6a68629aaba8556cf0b96c7fb47dbe97e9ac0323L1832-L1834. 

And if we look at the definition of these functions https://github.com/cockroachdb/cockroach/blob/37223e4bfcceb0b34bbf6879be39ff483b15ffd4/pkg/kv/db.go#L57-L88
`PrettyValue` prints the `RawBytes` if the value is a tuple (that is none of the primitive types as it is a combination of more than one columns). Whereas https://github.com/cockroachdb/cockroach/blob/37223e4bfcceb0b34bbf6879be39ff483b15ffd4/pkg/roachpb/data.go#L822-L903 does more hard work even for the value type of `ValueType_TUPLE` and print a more friendly output for all columns.

So here is what I think was happening before this whole behavior was fixed,

First, when `list_sql_keys_in_range` was used which eventually called `PrettyValue` with the `Value` of tuple type as it's a record of multiple columns, it printed the `RawBytes`.

Later when `column_b` is moved to a new family it is stored as a separate key-value pair, so for the default column family `Value` is still a tuple but for `new_family`'s key value pairs (the alternative rows in the output) the `Value` is just a string (`ValueType_BYTES`) and is printed in human-readable form.

And when `column_c` is also moved to `new_family`, `Value`s for `new_family` are also now a tuple and hence printed as `RawBytes`.

Please note I wanted to confirm this by putting some flags in the code but building two years old version is throwing some issue. So I would appreciate if someone can share their opinion on this analysis.

<details><summary>Here is the output on current `master`</summary>

```
demo@127.0.0.1:26257/demoapp/defaultdb> CREATE TABLE public.mytable(
                                     ->     id uuid NOT NULL,
                                     ->     column_a STRING NULL,
                                     ->     column_b STRING NULL,
                                     ->     column_c STRING NULL,
                                     ->     CONSTRAINT mytable_pkey PRIMARY KEY (id ASC)
                                     -> );
CREATE TABLE

Time: 4ms total (execution 4ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> INSERT INTO mytable
                                     -> SELECT
                                     ->     gen_random_uuid(),
                                     ->     'text for column a',
                                     ->     'text for column b',
                                     ->     'text for column c'
                                     -> FROM
                                     ->     generate_series(1, 3);
INSERT 0 3

Time: 5ms total (execution 5ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT range_id, start_key, end_key, replicas, lease_holder FROM [SHOW RANGES FROM
                                     -> TABLE mytable WITH DETAILS];
  range_id |  start_key   |      end_key      | replicas | lease_holder
-----------+--------------+-------------------+----------+---------------
        81 | …/<TableMin> | <after:/Tenant/3> | {1}      |            1
(1 row)

Time: 19ms total (execution 19ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT
                                     ->     *
                                     -> FROM
                                     ->     crdb_internal.list_sql_keys_in_range(81);
                                     key                                     |                                           value                                            |           ts
-----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+-------------------------
  /Tenant/2/Table/104/1/"J7\x83\x04'\xb6Mۧv\xc0\"\xb4\xba\x02#"/0             | /TUPLE/2:2:Bytes/text for column a/1:3:Bytes/text for column b/1:4:Bytes/text for column c | 1720106439.784367000,0
  /Tenant/2/Table/104/1/"\\\b\xe2\x86e\xc2J~\x93\xcf\xd0F\x93\x14\xfb\xfb"/0 | /TUPLE/2:2:Bytes/text for column a/1:3:Bytes/text for column b/1:4:Bytes/text for column c | 1720106439.784367000,0
  /Tenant/2/Table/104/1/"y\x949\x94\xb6\xfbK\u0089&\x91*.|\x8e\x8c"/0        | /TUPLE/2:2:Bytes/text for column a/1:3:Bytes/text for column b/1:4:Bytes/text for column c | 1720106439.784367000,0
(3 rows)

Time: 3ms total (execution 3ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     ADD COLUMN new_column string CREATE FAMILY "new_family" NOT VISIBLE AS (
                                     ->     column_b) STORED;
ALTER TABLE

Time: 181ms total (execution 181ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     ALTER COLUMN new_column
                                     ->     DROP STORED,
                                     ->     ALTER COLUMN new_column SET VISIBLE, RENAME COLUMN column_b TO old_column,
                                     -> RENAME COLUMN new_column TO column_b,
                                     ->     ALTER COLUMN old_column SET NOT VISIBLE;
ALTER TABLE

demo@127.0.0.1:26257/demoapp/defaultdb> set sql_safe_updates = false;
SET

Time: 1ms total (execution 1ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     DROP COLUMN old_column;
                                     ->
ALTER TABLE

Time: 183ms total (execution 183ms / network 0ms)

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT range_id, start_key, end_key, replicas, lease_holder FROM [SHOW RANGES FROM
                                     -> TABLE mytable WITH DETAILS];
  range_id |   start_key   |      end_key      | replicas | lease_holder
-----------+---------------+-------------------+----------+---------------
        81 | …/<TableMin>  | …/Table/104/2     | {1}      |            1
        82 | …/Table/104/2 | …/Table/104/3     | {1}      |            1
        83 | …/Table/104/3 | …/Table/104/4     | {1}      |            1
        84 | …/Table/104/4 | …/Table/104/5     | {1}      |            1
        85 | …/Table/104/5 | <after:/Tenant/3> | {1}      |            1
(5 rows)

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT * FROM crdb_internal.list_sql_keys_in_range(84);
                                      key                                      |                             value                              |           ts
-------------------------------------------------------------------------------+----------------------------------------------------------------+-------------------------
  /Tenant/2/Table/104/4/"J7\x83\x04'\xb6Mۧv\xc0\"\xb4\xba\x02#"/0               | /TUPLE/2:2:Bytes/text for column a/2:4:Bytes/text for column c | 1720106494.477737000,0
  /Tenant/2/Table/104/4/"J7\x83\x04'\xb6Mۧv\xc0\"\xb4\xba\x02#"/1/1             | /BYTES/text for column b                                       | 1720106494.477737000,0
  /Tenant/2/Table/104/4/"\\\b\xe2\x86e\xc2J~\x93\xcf\xd0F\x93\x14\xfb\xfb"/0   | /TUPLE/2:2:Bytes/text for column a/2:4:Bytes/text for column c | 1720106494.477737000,0
  /Tenant/2/Table/104/4/"\\\b\xe2\x86e\xc2J~\x93\xcf\xd0F\x93\x14\xfb\xfb"/1/1 | /BYTES/text for column b                                       | 1720106494.477737000,0
  /Tenant/2/Table/104/4/"y\x949\x94\xb6\xfbK\u0089&\x91*.|\x8e\x8c"/0          | /TUPLE/2:2:Bytes/text for column a/2:4:Bytes/text for column c | 1720106494.477737000,0
  /Tenant/2/Table/104/4/"y\x949\x94\xb6\xfbK\u0089&\x91*.|\x8e\x8c"/1/1        | /BYTES/text for column b                                       | 1720106494.477737000,0
(6 rows)

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     ADD COLUMN new_column string FAMILY "new_family" NOT VISIBLE AS (
                                     -> column_c) STORED;
ALTER TABLE

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     ALTER COLUMN new_column
                                     ->     DROP STORED,
                                     ->     ALTER COLUMN new_column SET VISIBLE, RENAME COLUMN column_c TO old_column,
                                     -> RENAME COLUMN new_column TO column_c,
                                     ->     ALTER COLUMN old_column SET NOT VISIBLE;
ALTER TABLE

demo@127.0.0.1:26257/demoapp/defaultdb> ALTER TABLE mytable
                                     ->     DROP COLUMN old_column;
ALTER TABLE

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT range_id, start_key, end_key, replicas, lease_holder FROM [SHOW RANGES FROM
                                     -> TABLE mytable WITH DETAILS];
  range_id |   start_key   |      end_key      | replicas | lease_holder
-----------+---------------+-------------------+----------+---------------
        81 | …/<TableMin>  | …/Table/104/2     | {1}      |            1
        82 | …/Table/104/2 | …/Table/104/3     | {1}      |            1
        83 | …/Table/104/3 | …/Table/104/4     | {1}      |            1
        84 | …/Table/104/4 | …/Table/104/5     | {1}      |            1
        85 | …/Table/104/5 | …/Table/104/6     | {1}      |            1
        86 | …/Table/104/6 | …/Table/104/7     | {1}      |            1
        87 | …/Table/104/7 | …/Table/104/8     | {1}      |            1
        88 | …/Table/104/8 | …/Table/104/9     | {1}      |            1
        89 | …/Table/104/9 | <after:/Tenant/3> | {1}      |            1
(9 rows)

demo@127.0.0.1:26257/demoapp/defaultdb> SELECT * FROM crdb_internal.list_sql_keys_in_range(88);
                                      key                                      |                             value                              |           ts
-------------------------------------------------------------------------------+----------------------------------------------------------------+-------------------------
  /Tenant/2/Table/104/8/"J7\x83\x04'\xb6Mۧv\xc0\"\xb4\xba\x02#"/0               | /TUPLE/2:2:Bytes/text for column a                             | 1720106550.888285000,0
  /Tenant/2/Table/104/8/"J7\x83\x04'\xb6Mۧv\xc0\"\xb4\xba\x02#"/1/1             | /TUPLE/5:5:Bytes/text for column b/1:6:Bytes/text for column c | 1720106550.888285000,0
  /Tenant/2/Table/104/8/"\\\b\xe2\x86e\xc2J~\x93\xcf\xd0F\x93\x14\xfb\xfb"/0   | /TUPLE/2:2:Bytes/text for column a                             | 1720106550.888285000,0
  /Tenant/2/Table/104/8/"\\\b\xe2\x86e\xc2J~\x93\xcf\xd0F\x93\x14\xfb\xfb"/1/1 | /TUPLE/5:5:Bytes/text for column b/1:6:Bytes/text for column c | 1720106550.888285000,0
  /Tenant/2/Table/104/8/"y\x949\x94\xb6\xfbK\u0089&\x91*.|\x8e\x8c"/0          | /TUPLE/2:2:Bytes/text for column a                             | 1720106550.888285000,0
  /Tenant/2/Table/104/8/"y\x949\x94\xb6\xfbK\u0089&\x91*.|\x8e\x8c"/1/1        | /TUPLE/5:5:Bytes/text for column b/1:6:Bytes/text for column c | 1720106550.888285000,0
(6 rows)

```
</details>

**Comment by shubhamdhama** [comment]
I'm closing this one as it's fixed.
